src/metabase/public_settings.clj

```diff
@@ -1059,11 +1059,22 @@ See [fonts](../configuring-metabase/fonts.md).")
   :default    false
   :type       :boolean)
 
+(defsetting attachment-row-limit
+  (deferred-tru "Row limit in file attachments excluding the header.")
+  :visibility :internal
+  :export?    true
+  :type       :positive-integer)
+
 (defsetting download-row-limit
-  (deferred-tru "Exports row limit excluding the header. xlsx downloads are limited to 1048575 rows even if this limit is higher.")
+  (deferred-tru "Row limit in file exports excluding the header. Enforces 1048575 excluding header as minimum. xlsx downloads are inherently limited to 1048575 rows even if this limit is higher.")
   :visibility :internal
   :export?    true
-  :type       :integer)
+  :type       :positive-integer
+  :getter     (fn []
+                (let [limit (setting/get-value-of-type :positive-integer :download-row-limit)]
+                  (if (nil? limit)
+                    limit
+                    (max limit 1048575)))))
```

Comment:
can use `qp.i/absolute-max-results` here instead of hard-coding the number again?

Reply:
or at the very least it could be defined as a dynamic var in this namespace like `*minimum-download-row-limit*` or something, then you can rebind it in the tests